lib/block-patterns.php

```diff
@@ -236,6 +236,58 @@ function load_remote_patterns() {
 	}
 }
 
+/**
+ * Register featured patterns from wordpress.org/patterns.
+ *
+ * @since 5.9.0
+ */
+function load_remote_featured_patterns() {
+	/**
+	 * Filter to disable remote block patterns.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool $should_load_remote
+	 */
+	$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+
+	if ( ! $should_load_remote ) {
+		return;
+	}
+
+	// TODO: check about the `featured` name.. Should this be reserved?
+	// Also check core logic: https://developer.wordpress.org/reference/classes/wp_block_pattern_categories_registry/
+	// Do we just allow collisions/overide with categories when they have the same name? Needs checking..
+	if ( ! WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'Featured' ) ) {
+		register_block_pattern_category( 'featured', array( 'label' => __( 'Featured', 'gutenberg' ) ) );
+	}
```

Comment:
It hasn't been fully implemented yet, but the pattern directory API was designed to [return a list of categories](https://api.wordpress.org/patterns/1.0/?categories) - eventually I think these should be dynamically registered, so we can add categories to the wporg site and have them show up automatically.

Reply:
This seems right! For now I'll just register the `featured` category.

Reply:
Sure! I was just saying for future reference, since you seem to be working on the pattern inserter :)